lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -204,11 +211,16 @@ newStakePoolLayer nl db@DBLayer {..} =
                 (nOpt rawLsqData)
                 lsqData
                 dbData
+        case r of
+            Left e@(PastHorizon{}) -> throwE (ErrListPoolsPastHorizonException e)
+            Right r' -> pure r'
```

Comment:
**Suggestion**: use the `first` combinator to avoid unnecessary pattern matching on `Right`.

```suggestion
        handlePastHorizon r
      where
        handlePastHorizon = except . first
            (\e@PastHorizon {} -> ErrListPoolsPastHorizonException e)
```

Reply:
why?

Reply:
> why?

BTW, this was just a suggestion, rather than a request. If you feel the original is clearer, then feel free to leave it that way.

My reasons for suggesting it:

* Using `first` allows you to omit the `Right` pattern match, which means that you can avoid the extra `r'` binding.
* The `first` and `second` functions (from `Data.Bifunctor`) are used fairly commonly in our code base, so it's a familiar pattern to the team.

Reply:
In this case it doesn't really matter, but I'd like to point out that these expressions aren't strictly the same:

1. pattern matching forces evaluation immediately, so there are slightly different strictness properties
2. depending on how you define the utility function `handlePastHorizon` it may or may not inline and (at worst) cause an additional closure allocation (in this case I believe it does inline, unless some mtl-ism blocks it from doing so)

This is why I'm usually a bit more conservative with rewriting expressions, because even if they look the same, they aren't always.